### DISCUSSION TITLE
How can you programmatically query the pipeline DAG

### DISCUSSION THREAD

**Discussion by huydinhle** [kubeflow/pipelines#10648] [General]
I tried to use both the python sdk and the mysql database to programmatically query  the dag of all pipelines we own. However, both of them doesn't have this information. Can someone point me to the right direction?

**Comment by rimolive** [comment]
Can you explain the use case? I'd say at first that you could do through k8s API by running `kubectl get workflow -A`, but all you get is the metadata, not the DAG graph which is rendered by KFP UI.
